src/scf_module.F90

```diff
@@ -661,6 +659,11 @@ subroutine scf(env, mol, wfn, basis, pcem, xtbData, solvation, &
    if (.not.pr.and.profile.and.minpr) &
       call timer%write_timing(env%unit,5,'SCC iter.')
    if (profile) call timer%measure(6,"molecular gradient")
+   
+   ! throw error for unconverged SCF
+   if (fail) then
+      call env%error("Self consistent charge iterator did not converge", source)
+   end if
```

Comment:
Let's move this error after the gradient calculation and before the printing. Some calculations access the environment and might abort the calculation if an error is present.